[Alerts] Lower DB access by caching events for inactive alerts

yanburman opened the pull request:
We want to lower performance footprint for alerts that are inactive. This comes with a cost of not persisting state between mlrun-api resets, but that is something we are OK with at this point

yanburman committed:
[Alerts] Lower DB access by caching events for inactive alerts

Signed-off-by: Yan Burman <yanburman@users.noreply.github.com>

liranbg reviewed (approved):
Approving. only concern is that its limit is subjected to the fact that if event has configuration of great counter, the `states` dictionary would become gigantic consuming memory.
@theSaarco fyi

yanburman committed:
[Alerts] Limit criteria count based on config

We want to limit count parameter values based on
config, so user can't abuse this value to crash chief

Signed-off-by: Yan Burman <yanburman@users.noreply.github.com>
